api/envoy/config/core/v3/base.proto

```diff
@@ -358,6 +358,17 @@ message HeaderValueOption {
   // existing values. Otherwise it replaces any existing values.
   // This field is deprecated and please use
   // :ref:`append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>` as replacement.
+  //
+  // .. attention::
+  //   The :ref:`ext_authz filter <envoy_v3_api_msg_extensions.filters.http.ext_authz.v3.ExtAuthz>` and
+  //   :ref:`ext_proc filter <envoy_v3_api_msg_extensions.filters.http.ext_proc.v3.ExternalProcessor>` have
+  //   error default value (false) for this field.
+  //   And the default value of
+  //   :ref:`new append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>`
+  //   is always APPEND_IF_EXISTS_OR_ADD.
+  //   It means if we remove this field, the default behavior in these filter will be changed and may break our
+  //   users.
+  //   So we should keep this field and couldn't remove it until the above issue is resolved one day.
```

Comment:
Emm...The last two sentences seems like a note for the developer, it shouldn't be part of the API document.

Reply:
As I mentioned in Slack, at this point the APIs are forever. We are never removing anything, so I think we can remove this. Can we rewrite this as end user documentation? Thank you.

/wait

Reply:
It's easy to do some comments update. But I think there still is some info gap. 🤣 I re-read the API version policy. There are some conclusions by my understanding. Feel free to point int out if I am wrong.

1. When we say `the APIs are forever`, it means that this field is forever in the `core.v3.HeaderValueOption`.
2. One day the v4 APIs is enabled, then this field will be removed from the `core.v4.HeaderValueOption`. But the break changes are acceptable for major API versions update.

Reply:
I haven't gotten around to doing an official policy update, but at this point realistically a) we are never doing v3 and b) we are never actually deleting any fields/code from Envoy. It's too disruptive. cc @envoyproxy/api-shepherds

Reply:
got it.